addons/project/models/project_project.py

```diff
@@ -242,10 +242,10 @@ def _compute_next_milestone_id(self):
             )
         }
         for project in self:
-            next_milestone = milestone_ids_per_project_id[project.id][:1]
-            project.next_milestone_id = next_milestone
-            project.can_mark_milestone_as_done = next_milestone.can_mark_milestone_as_done
-            project.is_milestone_deadline_exceeded = next_milestone.is_milestone_deadline_exceeded
+            milestone = milestone_ids_per_project_id.get(project.id)
+            project.next_milestone_id = milestone and milestone[:1]
+            project.can_mark_milestone_as_done =  milestone and milestone[:1].is_deadline_exceeded
+            project.is_milestone_deadline_exceeded =  milestone and milestone[:1].is_deadline_exceeded
```

Comment:
Alternatively, we could do something like that: 

```suggestion
            milestone = milestone_ids_per_project_id.get(project.id, self.env['project.milestone'])
            project.next_milestone_id = milestone[:1]
            project.can_mark_milestone_as_done = milestone.can_be_marked_as_done
            project.is_milestone_deadline_exceeded = milestone.is_deadline_exceeded
```

Reply:
> Alternatively, we could do something like that:

Done